### PR TITLE
dialects: Add strided layout attribute

### PR DESCRIPTION
**Pull request by math-fehr**
This PR adds the strided layout attribute (https://mlir.llvm.org/docs/Dialects/Builtin/#stridedlayoutattr),
as well as its parsing.

The parsing for the attribute inside a `memref` will be added in a later PR.

This PR depends on #639

### CODE REVIEW

**Review by PapyChacal — APPROVED** [review]
Yay, thanks a lot ☺️
LGTM!

**Review by georgebisbas — APPROVED** [review]

File: xdsl/parser.py [hunk 1]
```diff
@@ -1233,6 +1233,82 @@ def try_parse_builtin_attr(self) -> Attribute | None:
...
 
+        self._synchronize_lexer_and_tokenizer()
+        if self._current_token.text == "strided":
+            strided = self.parse_strided_layout_attr()
+            self._synchronize_lexer_and_tokenizer()
+            return strided
+
+        return None
+
+    def _parse_int_or_question(self,
```
Comment (webmiche) [inline xdsl/parser.py:1244]: This feels like a rather particular function. Does it make sense to have this? Wouldn't a parametrized version make more sense?

**Review by webmiche — APPROVED** [review]
LGTM, just two cents on parsing/naming.

**Review by AntonLydike — APPROVED** [review]
LGTM, only one minor question

**Review by superlopuh — APPROVED** [review]

File: xdsl/parser.py (hunk 1 above)
Comment (math-fehr) [reply xdsl/parser.py:1323]: This is not really used anywhere else, which is why I made this private for now.
The other solution is to make this a nested function, but I don't really like it, what do you think?

File: xdsl/printer.py [hunk 2]
```diff
@@ -477,6 +478,24 @@ def print_one_elem(val: Attribute):
...
+            self.print_list(attribute.strides.data, print_int_or_question,
+                            ', ')
+            self.print(']')
+            if attribute.offset == IntAttr(0):
+                self.print('>')
+                return
+            self.print(', offset: ')
+            print_int_or_question(attribute.offset)
+            self.print('>')
+            return
```
[the remark this answers is not in the document]
Comment (math-fehr) [reply xdsl/printer.py:497]: No, since it is a builtin attribute :/

**Pull request merged**